### ISSUE TITLE
Crash when deleting conversation

### ISSUE THREAD

**Issue by Moret84** [moezbhatti/qksms#829]
Hi I have a Moto X Play (lux) and the app is crashing whenever I delete a conversation. 
Reopening the app does not show the conversation, though. 

Log:
```
Could not set socket write timeout:
01-10 21:04:13.082  4148  9485 W Conscrypt:    java.lang.reflect.Method.invoke(Native Method)
01-10 21:04:13.082  4148  9485 W Conscrypt:    com.google.android.gms.org.conscrypt.Platform.setSocketWriteTimeout(:com.google.android.gms@11951438:13)
01-10 21:04:13.110  1773  4268 W InputMethodManagerService: Window already focused, ignoring focus gain of: com.android.internal.view.IInputMethodClient$Stub$Proxy@f0fbccd attribute=null, token = android.os.BinderProxy@8fba93b
01-10 21:04:13.143  4148  9485 W Conscrypt: Could not set socket write timeout:
01-10 21:04:13.143  4148  9485 W Conscrypt:    java.lang.reflect.Method.invoke(Native Method)
01-10 21:04:13.143  4148  9485 W Conscrypt:    com.google.android.gms.org.conscrypt.Platform.setSocketWriteTimeout(:com.google.android.gms@11951438:13)
01-10 21:04:13.432 23373 23415 D OpenGLRenderer: endAllActiveAnimators on 0x91dbd200 (MenuPopupWindow$MenuDropDownListView) with handle 0x969112f0
01-10 21:04:13.886  3948 27170 W Conscrypt: Could not set socket write timeout:
01-10 21:04:13.886  3948 27170 W Conscrypt:    java.lang.reflect.Method.invoke(Native Method)
01-10 21:04:13.886  3948 27170 W Conscrypt:    com.google.android.gms.org.conscrypt.Platform.setSocketWriteTimeout(:com.google.android.gms@11951438:13)
01-10 21:04:13.933  3948 27170 W Conscrypt: Could not set socket write timeout:
01-10 21:04:13.933  3948 27170 W Conscrypt:    java.lang.reflect.Method.invoke(Native Method)
01-10 21:04:13.933  3948 27170 W Conscrypt:    com.google.android.gms.org.conscrypt.Platform.setSocketWriteTimeout(:com.google.android.gms@11951438:13)
01-10 21:04:14.063  1773  2561 D MotoSensors: ALS 16
01-10 21:04:14.269 23373 23373 E REALM_JNI: Row -1824795472 is no longer attached!
01-10 21:04:14.269 23373 23373 E REALM_JNI: jni: ThrowingException 8, Object is no longer valid to operate on. Was it deleted by another thread?, .
01-10 21:04:14.269 23373 23373 E REALM_JNI: Exception has been thrown: Object is no longer valid to operate on. Was it deleted by another thread?
01-10 21:04:14.270 23373 23373 W System.err: io.reactivex.exceptions.OnErrorNotImplementedException: Object is no longer valid to operate on. Was it deleted by another thread?
01-10 21:04:14.275 23373 23373 W System.err: 	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.observers.SerializedObserver.onError(SerializedObserver.java:155)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.observable.ObservableWithLatestFrom$WithLatestFromObserver.onNext(ObservableWithLatestFrom.java:79)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.observable.ObservableCombineLatest$LatestCoordinator.drain(ObservableCombineLatest.java:209)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.observable.ObservableCombineLatest$LatestCoordinator.innerNext(ObservableCombineLatest.java:238)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.observable.ObservableCombineLatest$CombinerObserver.onNext(ObservableCombineLatest.java:307)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.observable.ObservableSwitchMap$SwitchMapObserver.drain(ObservableSwitchMap.java:276)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.observable.ObservableSwitchMap$SwitchMapInnerObserver.onNext(ObservableSwitchMap.java:328)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.observable.ObservableFromPublisher$PublisherSubscriber.onNext(ObservableFromPublisher.java:56)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.flowable.FlowableCreate$LatestAsyncEmitter.drain(FlowableCreate.java:683)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.reactivex.internal.operators.flowable.FlowableCreate$LatestAsyncEmitter.onNext(FlowableCreate.java:609)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.realm.rx.RealmObservableFactory$6$1.onChange(RealmObservableFactory.java:161)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.realm.rx.RealmObservableFactory$6$1.onChange(RealmObservableFactory.java:157)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.realm.internal.ObservableCollection$RealmChangeListenerWrapper.onChange(ObservableCollection.java:40)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.realm.internal.ObservableCollection$CollectionObserverPair.onChange(ObservableCollection.java:21)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.realm.internal.ObservableCollection$Callback.onCalled(ObservableCollection.java:65)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.realm.internal.ObservableCollection$Callback.onCalled(ObservableCollection.java:55)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.realm.internal.ObserverPairList.foreach(ObserverPairList.java:109)
01-10 21:04:14.276 23373 23373 W System.err: 	at io.realm.internal.OsResults.notifyChangeListeners(OsResults.java:442)
01-10 21:04:14.276 23373 23373 W System.err: 	at android.os.MessageQueue.nativePollOnce(Native Method)
01-10 21:04:14.276 23373 23373 W System.err: 	at android.os.MessageQueue.next(MessageQueue.java:323)
01-10 21:04:14.276 23373 23373 W System.err: 	at android.os.Looper.loop(Looper.java:136)
01-10 21:04:14.276 23373 23373 W System.err: 	at android.app.ActivityThread.main(ActivityThread.java:6186)
01-10 21:04:14.276 23373 23373 W System.err: 	at java.lang.reflect.Method.invoke(Native Method)
01-10 21:04:14.276 23373 23373 W System.err: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
01-10 21:04:14.276 23373 23373 W System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
01-10 21:04:14.277 23373 23373 W System.err: Caused by: java.lang.IllegalStateException: Object is no longer valid to operate on. Was it deleted by another thread?
01-10 21:04:14.277 23373 23373 W System.err: 	at io.realm.internal.UncheckedRow.nativeGetLong(Native Method)
01-10 21:04:14.277 23373 23373 W System.err: 	at io.realm.internal.UncheckedRow.getLong(UncheckedRow.java:132)
01-10 21:04:14.277 23373 23373 W System.err: 	at io.realm.ConversationRealmProxy.realmGet$id(ConversationRealmProxy.java:106)
01-10 21:04:14.277 23373 23373 W System.err: 	at data.model.Conversation.getId(Conversation.kt:27)
01-10 21:04:14.277 23373 23373 W System.err: 	at presentation.feature.compose.ComposeViewModel$bindView$$inlined$withLatestFrom$4.apply(Observables.kt:192)
01-10 21:04:14.277 23373 23373 W System.err: 	at io.reactivex.internal.operators.observable.ObservableWithLatestFrom$WithLatestFromObserver.onNext(ObservableWithLatestFrom.java:75)
01-10 21:04:14.277 23373 23373 W System.err: 	... 23 more
01-10 21:04:14.295  2924  2924 D wpa_supplicant: wlan0: Control interface command 'SIGNAL_POLL'
01-10 21:04:14.345 23373 23400 W System  : ClassLoader referenced unknown path: /system/framework/tcmclient.jar
01-10 21:04:14.536 23373 23373 E AndroidRuntime: FATAL EXCEPTION: main
01-10 21:04:14.536 23373 23373 E AndroidRuntime: Process: com.moez.QKSMS, PID: 23373
01-10 21:04:14.536 23373 23373 E AndroidRuntime: io.reactivex.exceptions.OnErrorNotImplementedException: Object is no longer valid to operate on. Was it deleted by another thread?
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.observers.SerializedObserver.onError(SerializedObserver.java:155)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.observable.ObservableWithLatestFrom$WithLatestFromObserver.onNext(ObservableWithLatestFrom.java:79)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.observable.ObservableCombineLatest$LatestCoordinator.drain(ObservableCombineLatest.java:209)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.observable.ObservableCombineLatest$LatestCoordinator.innerNext(ObservableCombineLatest.java:238)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.observable.ObservableCombineLatest$CombinerObserver.onNext(ObservableCombineLatest.java:307)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.observable.ObservableSwitchMap$SwitchMapObserver.drain(ObservableSwitchMap.java:276)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.observable.ObservableSwitchMap$SwitchMapInnerObserver.onNext(ObservableSwitchMap.java:328)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.observable.ObservableFromPublisher$PublisherSubscriber.onNext(ObservableFromPublisher.java:56)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.flowable.FlowableCreate$LatestAsyncEmitter.drain(FlowableCreate.java:683)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.flowable.FlowableCreate$LatestAsyncEmitter.onNext(FlowableCreate.java:609)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.rx.RealmObservableFactory$6$1.onChange(RealmObservableFactory.java:161)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.rx.RealmObservableFactory$6$1.onChange(RealmObservableFactory.java:157)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.internal.ObservableCollection$RealmChangeListenerWrapper.onChange(ObservableCollection.java:40)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.internal.ObservableCollection$CollectionObserverPair.onChange(ObservableCollection.java:21)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.internal.ObservableCollection$Callback.onCalled(ObservableCollection.java:65)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.internal.ObservableCollection$Callback.onCalled(ObservableCollection.java:55)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.internal.ObserverPairList.foreach(ObserverPairList.java:109)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.internal.OsResults.notifyChangeListeners(OsResults.java:442)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at android.os.MessageQueue.nativePollOnce(Native Method)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at android.os.MessageQueue.next(MessageQueue.java:323)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:136)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6186)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: Caused by: java.lang.IllegalStateException: Object is no longer valid to operate on. Was it deleted by another thread?
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.internal.UncheckedRow.nativeGetLong(Native Method)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.internal.UncheckedRow.getLong(UncheckedRow.java:132)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.realm.ConversationRealmProxy.realmGet$id(ConversationRealmProxy.java:106)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at data.model.Conversation.getId(Conversation.kt:27)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at presentation.feature.compose.ComposeViewModel$bindView$$inlined$withLatestFrom$4.apply(Observables.kt:192)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	at io.reactivex.internal.operators.observable.ObservableWithLatestFrom$WithLatestFromObserver.onNext(ObservableWithLatestFrom.java:75)
01-10 21:04:14.536 23373 23373 E AndroidRuntime: 	... 23 more
01-10 21:04:14.543  1773  4276 W ActivityManager:   Force finishing activity com.moez.QKSMS/presentation.feature.compose.ComposeActivity
```

**Comment by moezbhatti** [comment]
Which version of the app are you using?

**Comment by Moret84** [comment]
Version 3.0 Preview 4

**Comment by moezbhatti** [comment]
Ah okay, I believe I know what the issue is.

The conversation does get deleted, but the UI isn't informed of this and it's still trying to access the deleted conversation. So then the app crashes, because it doesn't exist.

I'll make sure to have this fixed in the next release. Thanks!

**Comment by moezbhatti** [comment]
I can't seem to reproduce this. Does it happen for you every time?

**Comment by Moret84** [comment]
Weirdly I tried multiple times but couldn't reproduce anymore.

Believe me, it did happen,though.But can't be reproduced each time so it will be a pain to fix.

**Comment by moezbhatti** [comment]
Ok. The reason I ask is because I used to have this issue as well, but I fixed it here: 7447ebc85fbe1d868511c54366afe35fc70c6a63

It's possible that you were on an earlier version when you saw the issue, but now you're using one where it's fixed.

I'll close the issue as I'm also not seeing any reports of this in the crash logs anymore, but if it pops up again feel free to ping me and I'll take another look!
